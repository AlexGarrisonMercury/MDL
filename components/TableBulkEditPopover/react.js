import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames/bind'
import styles from './style.module.scss'

const cx = classNames.bind(styles)

const TableBulkEditPopover = ({ children, direction, cell }) => (
  <th
    style={{ position: 'absolute' }}
    className={cx('popover-wrapper', {
      'popover-wrapper--top': direction === 'top',
      'popover-wrapper--left': direction === 'left',
    })}
  >
    <div
      className={cx('popover', {
        'popover--top': direction === 'top',
        'popover--left': direction === 'left',
      })}
    >
      {children}
    </div>
    <div className={cx('popover-wrapper__cell')}>{cell}</div>
  </th>
)

export default TableBulkEditPopover

TableBulkEditPopover.propTypes = {
  children: PropTypes.oneOfType([PropTypes.element, PropTypes.string]).isRequired,
  direction: PropTypes.string.isRequired,
  cell: PropTypes.string.isRequired,
}
