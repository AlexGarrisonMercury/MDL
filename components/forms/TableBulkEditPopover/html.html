<th class="popover-wrapper">
  <div className= "popover-wrapper--position-top">
    <TextField />
  </div>
  <div className="popover-wrapper--cell">4'2"</div>
</th>
