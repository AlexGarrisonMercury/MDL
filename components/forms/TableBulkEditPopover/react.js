import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames/bind'
import styles from './style.module.scss'

const cx = classNames.bind(styles)

const TableBulkEditPopover = ({ children, direction, cell }) => (
  <th class={cx('popover-wrapper', {
    'popover-wrapper--top': direction === 'top',
    'popover-wrapper--left': direction === 'left',
  })}
  >
    <div className={cx({

      'popover-wrapper--position-top': direction === 'top',
      'popover-wrapper--position-left': direction === 'left',

    })}>
      {children}
    </div>
    <div className={cx('popover-wrapper--cell')}>{cell}</div>
  </th>
)

export default TableBulkEditPopover

TableBulkEditPopover.propTypes = {
  children: PropTypes.oneOfType([PropTypes.element, PropTypes.string]).isRequired,
  direction: PropTypes.string.isRequired
}
