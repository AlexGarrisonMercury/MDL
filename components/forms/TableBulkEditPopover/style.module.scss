@import "styles/index";

.popover-wrapper {
  .popover {
    position: absolute;
    background: $color-night;
    padding: $space-xs;

    &--top {
      bottom: 100%;
    }

    &--left {
      right: 100%;
    }
  }

  &__cell {
    border: $color-night solid 1px;
  }
}
