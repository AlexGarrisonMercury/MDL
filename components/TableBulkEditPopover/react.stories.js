import React from 'react'
import { storiesOf } from '@storybook/react'
import TableBulkEditPopover from './react'
import { React as TextField } from '../TextField'
import Readme from './react.README.md'

const styles = {
  width: '40px',
}

storiesOf('TableBulkEditPopover', module)
  .addParameters({ readme: { sidebar: Readme } })
  .add('example', () => (
    <div style={{ marginLeft: 100, marginTop: 100 }}>
      <TableBulkEditPopover direction="left" cell={'4\'2"'}>
        <div style={styles}>
          <TextField onChange={() => { }} value="" variant="light" removeMargin="true" />
        </div>
      </TableBulkEditPopover>
    </div>
  ))
