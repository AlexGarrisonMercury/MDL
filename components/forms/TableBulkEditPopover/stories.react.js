import React from 'react'
import { storiesOf } from '@storybook/react'
import TableBulkEditPopover from './react'
import { React as TextField } from '../TextField'

let styles = {
 width:'40px',
 top:'5px',
 left:'6px',
 position:'relative'
};

storiesOf('TableBulkEditPopover', module).add('example', () => (
  <TableBulkEditPopover direction="left" cell={'4\'2"'}>
    <div style={styles}>
      <TextField onChange={() => { }} value="" variant="light" />
    </div>
  </TableBulkEditPopover>
))