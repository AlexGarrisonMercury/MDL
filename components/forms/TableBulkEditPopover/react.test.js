import React from 'react'
import { shallow } from 'enzyme'
import TableBulkEditPopover from './react'

describe('TableBulkEditPopover', () => {
  let props
  let wrapper

  beforeEach(() => {
    props = {
     direction: 'left',
     childern: 'field',
     cell: '4.2'
    }

    wrapper = shallow(<TableBulkEditPopover {...props} />)
   
  })

  describe('rendering with defaults', () => {
    it('should render correctly', () => {
      expect(wrapper).toMatchSnapshot()
    })

    it('should return cell value',()=>{
     expect (wrapper.find('.popover-wrapper--cell').text()).toBe('4.2')
    })
    it('should return left direction class',()=>{
      expect (wrapper.find('.popover-wrapper--position-left').length).toBe(1)
    })

    it('should return top direction class',()=>{
      props = {
        ...props,
        direction: 'top',
       }
   
       wrapper = shallow(<TableBulkEditPopover {...props} />)
      expect (wrapper.find('.popover-wrapper--position-top').length).toBe(1)
    })
  })
})
