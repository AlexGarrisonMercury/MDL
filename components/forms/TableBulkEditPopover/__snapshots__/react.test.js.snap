// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableBulkEditPopover rendering with defaults should render correctly 1`] = `
<th
  class="popover-wrapper popover-wrapper--left"
>
  <div
    className="popover-wrapper--position-left"
  />
  <div
    className="popover-wrapper--cell"
  >
    4.2
  </div>
</th>
`;
